marklogic-client-api/src/main/java/com/marklogic/client/impl/DocumentWriteOperationImpl.java

```diff
@@ -70,4 +81,37 @@ public DocumentMetadataWriteHandle getMetadata() {
   public AbstractWriteHandle getContent() {
     return content;
   }
+
+  @Override
+  public int compareTo(DocumentWriteOperation o) {
+    if(o == null)
+      throw new NullPointerException("DocumentWriteOperation cannot be null");
+
+    if(this.getUri() instanceof String && o.getUri() instanceof String)
```

Comment:
instanceof implies that a specific subtype instead of null is being checked.

Would != null be easier to understand relative to the == null tests?